config/routes.rb

```diff
@@ -28,6 +28,12 @@
     resources  :categories, only: :index
     resources  :prospective_users, only: :create
     resources  :assessments, only: [:index, :show, :update, :create] do
+      collection do
+        resources :shared, controller: 'shared_assessments', only: [:show], param: :token do
```

Comment:
Like I said, I'm not bothered with the way it is now, but I'm not necessarily opposed to the repetition.  Since we're using three different controllers, the only thing they have in common is the path which you'd get with `:shared`.